src/main/java/org/spongepowered/common/mixin/core/entity/ai/MixinEntityAIMoveToBlock.java

```diff
@@ -0,0 +1,109 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.mixin.core.entity.ai;
+
+import com.flowpowered.math.vector.Vector3i;
+import net.minecraft.entity.ai.EntityAIMoveToBlock;
+import net.minecraft.util.math.BlockPos;
+import org.spongepowered.api.entity.ai.task.builtin.creature.location.MoveToBlockAITask;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.common.interfaces.ai.IMixinEntityAIMoveToBlock;
+
+import java.util.function.Predicate;
+
+@Mixin(EntityAIMoveToBlock.class)
+public abstract class MixinEntityAIMoveToBlock extends MixinEntityAIBase implements MoveToBlockAITask, IMixinEntityAIMoveToBlock {
+
+    @Shadow public int searchLength;
+    @Shadow public double movementSpeed;
+    @Shadow private int timeoutCounter;
+
+    @Shadow public abstract void updateTask();
+    @Shadow protected abstract boolean shouldMoveTo(net.minecraft.world.World worldIn, BlockPos pos);
+
+    @Shadow protected BlockPos destinationBlock;
+
+    private Predicate<Location<World>> destinationPredicate;
+    private boolean vanillaTask = false;
+
+    @Override
+    public boolean isVanillaTask() {
+        return this.vanillaTask;
+    }
+
+    @Override
+    public double getSpeed() {
+        return this.movementSpeed;
+    }
+
+    @Override
+    public MoveToBlockAITask setSpeed(double speed) {
+        this.movementSpeed = speed;
+        return this;
+    }
+
+    @Override
+    public int getSearchRange() {
+        return this.searchLength;
+    }
+
+    @Override
+    public MoveToBlockAITask setSearchRange(int searchRange) {
+        this.searchLength = searchRange;
+        return this;
+    }
+
+    @Override
+    public Predicate<Location<World>> getDestinationPredicate() {
+        if (this.destinationPredicate == null) {
+            this.vanillaTask = true;
+            this.destinationPredicate = worldLocation ->
+                    shouldMoveTo((net.minecraft.world.World) worldLocation.getExtent(),
+                        new BlockPos(worldLocation.getX(), worldLocation.getY(), worldLocation.getZ()));
+        }
+        return this.destinationPredicate;
+    }
+
+    @Override
+    public void setDestinationPredicate(Predicate<Location<World>> destinationPredicate) {
+        this.destinationPredicate = destinationPredicate;
+    }
+
+    @Override
+    public Vector3i getDestination() {
+        return new Vector3i(destinationBlock.getX(), destinationBlock.getY(), destinationBlock.getZ());
+    }
+
+    @Override
+    public MoveToBlockAITask setDestination(Vector3i destination) {
+        this.destinationBlock = new BlockPos(destination.getX(), destination.getY(), destination.getZ());
+        this.timeoutCounter = 39;
+        this.updateTask();
+        return this;
+    }
+}
```

Comment:
So on my local machine there's a line there - not sure why it isn't there on github. I'll look into it.